docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,381 @@
+---
+title: New Behavior
+sidebar_label: New Behavior
+---
+
+## Overview
+
+This document outlines how to develop a behavior for ZMK and prepare the changes for a pull request.
+
+The general process for developing behaviors is:
+
+1. [Create the behavior](#creating-the-behavior)
+   1. [Create the devicetree binding (`.yaml`)](#creating-the-devicetree-binding-yaml)
+   1. [Create the driver (`.c`)](#creating-the-driver-c)
+   1. [Update `app/CmakeLists.txt` to include the new driver](#updating-appcmakeliststxt-to-include-the-new-driver)
+   1. [Define common use-cases for the behavior (`.dtsi`) (Optional)](#defining-common-use-cases-for-the-behavior-dtsi-optional)
+1. [Test changes locally](#testing-changes-locally)
+1. [Document behavior functionality](#documenting-behavior-functionality)
+1. [Create a Pull Request for review and inclusion into the ZMK sources](#submitting-a-pull-request)
+
+:::info
+Before developing new behaviors, developers should have a working knowledge of the Embedded Linux Devicetree.
+The following resources are provided for those seeking further understanding:
+
+- [Embedded Linux Wiki - Device Tree Usage](https://elinux.org/Device_Tree_Usage)
+- [Zephyr Devicetree API](https://docs.zephyrproject.org/latest/build/dts/api/api.html)
+- [Zephyr Device Driver Model](https://docs.zephyrproject.org/latest/kernel/drivers/index.html)
+
+:::
+
+## Creating the Behavior
+
+### Creating the devicetree binding (`.yaml`)
+
+The properties of the behavior are listed in the behavior's devicetree binding, which comes in the form of a `.yaml` file. Devicetree bindings are stored in the directory `app/dts/bindings/behaviors/` and are labelled in lowercase, beginning with the prefix `zmk,behavior-`, and ending with the behavior's name, using dashes to separate multiple words. For example, the directory for the hold-tap's devicetree binding would be located at `app/dts/bindings/behaviors/zmk,behavior-hold-tap.yaml`, which is shown below as a reference:
+
+```
```

Comment:
Speaking as a total outsider: Is it worth including syntax directives in the code snippets? At a glance the docs site doesn't have these, but Docusaurus does support it (as does GitHub).

````
```yaml
# SPDX-License-Identifier: MIT
description: Hold or Tap behavior
```
````

yields

```yaml
# SPDX-License-Identifier: MIT
description: Hold or Tap behavior
```

Reply:
The syntax-based colour-coding is definitely a nice indication for developers using this guide to confirm which parts should be edited versus those that should be left as-is. I'd be happy to implement the syntax directives in the upcoming commits. 👍

Reply:
Definitely. I don’t see devicetree in [the list of supported languages for syntax highlighting](https://docusaurus.io/docs/markdown-features/code-blocks#supported-languages), but it would be nice to have highlighting wherever else possible. (We could maybe look to enable devicetree support for our docs in a separate PR?) 

Here’s the Docusaurus docs: [Code blocks](https://docusaurus.io/docs/markdown-features/code-blocks).

Also nice is [code titles](https://docusaurus.io/docs/markdown-features/code-blocks#code-title) and [line highlighting](https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting).